### PR TITLE
Jpa

### PR DESCRIPTION
**Pull request by Stipaxa-hub**

### CODE REVIEW

File: src/main/java/mate/project/store/entity/Book.java [hunk 1]
```diff
@@ -5,25 +5,43 @@
...
+import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
+@Table(name = "books")
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
Comment (okuzan) [inline src/main/java/mate/project/store/entity/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
@Where(clause = "is_deleted = FALSE")
```

File: src/main/java/mate/project/store/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
+        return bookMapper.toDto(bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id " + id)));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto updateById(Long id, CreateBookRequestDto bookRequestDto) {
+        bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find a book with id " + id));
```
Comment (okuzan) [inline src/main/java/mate/project/store/service/impl/BookServiceImpl.java]: create a private method for find or throw logic

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: make sure each file ends with exactly one empty line
image

File: src/main/java/mate/project/store/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+
```
Comment (okuzan) [inline src/main/java/mate/project/store/controller/BookController.java:23]: no empty line after class definition

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**